eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -817,6 +817,15 @@ class Channel(val nodeParams: NodeParams, wallet: EclairWallet, remoteNodeId: Pu
       val channelUpdate = Announcements.makeChannelUpdate(nodeParams.chainHash, nodeParams.privateKey, remoteNodeId, d.shortChannelId, d.channelUpdate.cltvExpiryDelta, d.channelUpdate.htlcMinimumMsat, d.channelUpdate.feeBaseMsat, d.channelUpdate.feeProportionalMillionths, enable = true)
       stay using store(d.copy(channelUpdate = channelUpdate))
 
+    case Event(CMD_UPDATE_RELAY_FEE(feeBaseMsat, feeProportionalMillionths), d: DATA_NORMAL) =>
+      // Command from API
+      log.debug(s"sending channel_update announcement (feeBaseMsat=$feeBaseMsat feeProportionalMillionths=$feeProportionalMillionths)")
+      val channelUpdate = Announcements.makeChannelUpdate(nodeParams.chainHash, nodeParams.privateKey, remoteNodeId, d.shortChannelId, d.channelUpdate.cltvExpiryDelta, d.channelUpdate.htlcMinimumMsat, feeBaseMsat, feeProportionalMillionths, enable = true)
+      context.system.eventStream.publish(LocalChannelUpdate(self, d.commitments.channelId, d.shortChannelId, d.commitments.remoteParams.nodeId, d.channelAnnouncement, channelUpdate))
```

Comment:
Tested with 4 nodes running in regtest mode and really need this line.
Without it updates only published after I bounced the node.

Reply:
It is because transitions are only triggered when using `goto`, but I see that you noticed that.

I'd rather use `goto(NORMAL) using store(d.copy(channelUpdate = channelUpdate)) replying "ok"` instead of `handleCommandSuccess`